scylla/src/client/session.rs

```diff
@@ -1199,6 +1199,14 @@ impl Session {
     /// Prepares a statement on the server side and returns a prepared statement,
     /// which can later be used to perform more efficient requests.
     ///
+    /// The statement is prepared on all nodes. This function finishes once all nodes respond
+    /// with either success or an error.
+    // TODO: There are no timeouts here. So, just one stuck node freezes the driver here, potentially indefinitely long.
+    // Also, what the driver requires to get from the cluster is the prepared statement metadata.
+    // It suffices that it gets only one copy of it, just from one success response. Therefore, it's a possible
+    // optimisation that the function only waits for one preparation to finish successfully, and then it returns.
+    // For it to be done, other preparations must continue in the background, on a separate tokio task.
+    ///
```

Comment:
The described idea is imo not a TODO, because it is not obvious it is something we should do. It should be an issue instead.

Reply:
Why shouldn't it be a TODO? Issues have the problem that they are not immediately visible when someone is reading the code. OTOH, such a TODO catches attention quickly and makes the reader aware of the described problem and considered solutions.